gpt_index/tools/migrate_v1_to_v2.py

```diff
@@ -0,0 +1,213 @@
+"""Tool for migrating Index built with V1 data structs to V2."""
+import json
+from typing import Dict, List, Optional, Tuple, Type
+
+try:
+    import fire
+except ImportError:
+    print('Please run `pip install fire`')
+
+from gpt_index.data_structs.data_structs import (
+    KG,
+    ChromaIndexDict,
+    EmptyIndex,
+    FaissIndexDict,
+    IndexDict,
+    IndexGraph,
+    IndexList,
+    IndexStruct,
+    KeywordTable,
+    Node,
+    PineconeIndexDict,
+    QdrantIndexDict,
+    SimpleIndexDict,
+    WeaviateIndexDict,
+)
+from gpt_index.data_structs.data_structs_v2 import KG as V2KG
+from gpt_index.data_structs.data_structs_v2 import IndexDict as V2IndexDict
+from gpt_index.data_structs.data_structs_v2 import IndexGraph as V2IndexGraph
+from gpt_index.data_structs.data_structs_v2 import IndexList as V2IndexList
+from gpt_index.data_structs.data_structs_v2 import KeywordTable as V2KeywordTable
+from gpt_index.data_structs.data_structs_v2 import V2IndexStruct
+from gpt_index.data_structs.node_v2 import DocumentRelationship
+from gpt_index.data_structs.node_v2 import Node as V2Node
+from gpt_index.data_structs.struct_type import IndexStructType
+from gpt_index.data_structs.table_v2 import SQLStructTable
+from gpt_index.docstore import DocumentStore
+from gpt_index.indices.registry import INDEX_STRUCT_TYPE_TO_INDEX_STRUCT_CLASS
+from gpt_index.tools.file_utils import add_prefix_suffix_to_file_path
+
+INDEX_STRUCT_TYPE_TO_INDEX_STRUCT_CLASS: Dict[IndexStructType, IndexStruct]= {
+    IndexStructType.TREE: IndexGraph,
+    IndexStructType.LIST: IndexList,
+    IndexStructType.KEYWORD_TABLE: KeywordTable,
+    IndexStructType.SIMPLE_DICT: SimpleIndexDict,
+    IndexStructType.DICT: FaissIndexDict,
+    IndexStructType.WEAVIATE: WeaviateIndexDict,
+    IndexStructType.PINECONE: PineconeIndexDict,
+    IndexStructType.QDRANT: QdrantIndexDict,
+    IndexStructType.CHROMA: ChromaIndexDict,
+    IndexStructType.VECTOR_STORE: IndexDict,
+    IndexStructType.SQL: SQLStructTable,
+    IndexStructType.KG: KG,
+    IndexStructType.EMPTY: EmptyIndex,
+    IndexStructType.NODE: Node
+}
+
+INDEX_STRUCT_KEY = "index_struct"
+INDEX_STRUCT_ID_KEY = "index_struct_id"
+DOC_STORE_KEY = "docstore"
+
+def node_to_v2(node: Node) -> V2Node:
+    return V2Node(
+        text=node.text,
+        doc_id=node.doc_id,
+        embedding=node.embedding,
+        doc_hash=node.doc_hash,
+        extra_info=node.extra_info,
+        node_info=node.node_info,
+        image=node.image,
+        relationships={
+            DocumentRelationship.SOURCE: node.ref_doc_id,
+        }
+    )
+
+def index_graph_to_v2(struct: IndexGraph) -> Tuple[V2IndexGraph, List[V2Node]]:
```

Comment:
is it possible to unit test each type? i guess, we don't have the old index classes anymore, but we could make up a "fake" docstore with each index struct type maybe.

i can help write

Reply:
Seems like a good idea, but don't think I can get to it tonight.